cmd/queue/main.go

```diff
@@ -188,19 +201,28 @@ func handler(reqChan chan queue.ReqEvent, breaker *queue.Breaker, handler http.H
 					// Respond with the name of the component handling the request.
 					w.Write([]byte(queue.Name))
 				} else {
+					probeSpan.Annotate([]trace.Attribute{
+						trace.StringAttribute("queueproxy.probe.error", "container not ready")}, "error")
+					probeSpan.End()
 					http.Error(w, "container not ready", http.StatusServiceUnavailable)
 				}
+
 			} else {
+				probeSpan.Annotate([]trace.Attribute{
+					trace.StringAttribute("queueproxy.probe.error", "no probe")}, "error")
+				probeSpan.End()
 				http.Error(w, "no probe", http.StatusInternalServerError)
 			}
 
 			return
 		case network.IsKubeletProbe(r):
 			// Do not count health checks for concurrency metrics
-			handler.ServeHTTP(w, r)
+			handler.ServeHTTP(w, r.WithContext(reqCtx))
```

Comment:
Is `probeSpan.End` missing?

Reply:
I think we should actually be okay to remove the other probeSpan.End(), and keep the one directly after the switch statement.

Reply:
I don't think that works for us. Note the early return statements in the switch.

Reply:
I think my update for this should work.